### ISSUE TITLE
Android & IOS Signature doesn't match

### ISSUE THREAD

**Issue by shaxxx** [Dviejopomata/flutter-p12#1]
I'm using my P12 cert to sign "Hello world" string in example app but results are not the same on IOS & Android. Signing same data ("Hello world") with same certificate (my P12 file) should produce identical signature on both IOS & Android using example app.

I did some testing and Android produces exactly same result as .NET test I did with same data and certificate. So this pinpointed problem to IOS. Problem is IOS implementation is not same as the one on Android (PKCSA/SHA1). 

I was able to modify IOS to give the same result as Android but wanted to have a feedback from you about the API. Currently PKCSA/SHA1 is hardcoded in Android, and IOS has no hash algorithm specified. If someone wants to use ie. SHA256 this has to be changed inside the plugin, there's no hash algorithm parameter.

Thanks for the plugin!

**Comment by shaxxx** [comment]
I decided to move it to new [plugin](https://github.com/shaxxx/flutter_pkcs12) to modify API to my needs. Once again, thanks for the great plugin.

**Comment by Dviejopomata** [comment]
Sorry shaxxx, didn't see the notification, 24 days ago omg, had the notifications in another email.

I saw the error, I think I didn't make the SHA before signing with PKCS1SHA1, my fault.

Going through the repository I saw some Spanish texts that should be English. Going to check out.

Thanks for reporting!

**Comment by shaxxx** [comment]
No problem, glad you responded. Like you said you'll need to hash it before signing (on native side to keep API same for iOS & Android) and use `SecPadding.PKCS1SHA1` instead of `SecPadding.PKCS1`.
I decided to go with new plugin mostly because you didn't respond and I needed to fix this and implement a way to use other hash types (ie. SHA256 & SHA512). Also, I decided to leave only methods fully implemented and available on both platforms (2 methods to be exact). For example, plugin should not be concerned how certificate will be provided, it should expect it as a parameter and leave it to client app to provide it. Using activities in plugin is bad idea IMHO.

Well, it was easy for me to reinvent the wheel once you created the plugin.
Once again, thanks for the great work, it's really appreciated.
